src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -99,7 +99,8 @@ protected override ImmutableArray<LocalSymbol> BuildLocals()
                 case SyntaxKind.IdentifierName:
                     break;
                 default:
-                    throw ExceptionUtilities.UnexpectedValue(declaration.Kind());
+                    ExpressionVariableFinder.FindExpressionVariables(this, locals, declaration);
```

Comment:
Sure. The tests would be `UnitTests.CodeGen.CodeGenDeconstructTests.MixedDeconstruction_06`, `UnitTests.CodeGen.CodeGenDeconstructTests.ForeachIntoExpression`, and `UnitTests.CodeGen.CodeGenDeconstructTests.MixedDeconstruction_05`. Comments in the next iteration.

---
In reply to: [91385737](https://github.com/dotnet/roslyn/pull/15548#discussion_r91385737) [](ancestors = 91385737)